Makefile

```diff
@@ -0,0 +1,12 @@
+.PHONY: all
+all: install test
+
+.PHONY: install
+install:
+	wget https://github.com/archiecobbs/logwarn/archive/refs/tags/1.0.17.tar.gz
+	tar -xzvf 1.0.17.tar.gz
+	pushd logwarn-1.0.17 && ./autogen.sh && ./configure --prefix=/usr && make && sudo make install
+
```

Comment:
The "install" target by convention is to install the source of the current project, *not* to install test dependencies. Please rename the make target. What about "install-deps"?

Reply:
Install targets are *not* for *installing* sources but @okurz 's point is nevertheless very important. I would refrain from`install` and also `install-something` completely. The latter is still usually for installing a special subset of files. Normally you just call a target by what the end product is (`install` is a bit of an exception). So what about just `logwarn`?

Another problem is the use of `sudo`. This is a no-go. I'd never call a build script as root on my production system. I'd also never invoke different build targets as multiple/different users as this often leads to problems (e.g. files created by root can later not written to anymore by the regular user). If the destination is not writable by the user used to build I highly recommend a staged installation (use `DESTDIR` and move the tree afterwards manually).

Hard-coding the prefix is also bad. What if I don't want this under `/usr`? I most certainly won't want to have it there as this is meant for files installed via packages.

It also seems problematic that this attempts the download, the extraction and configuration even if even if that's already been done. This whole helper seems like a one-time setup. This is nothing I'd add to a Makefile as normally the expectation with Makefile targets is that they only do something if it hasn't already been done.

Reply:
yeah, I meant "install from source", not "install the source"